src/mapgen/mapgen_flat.cpp

```diff
@@ -52,17 +53,21 @@ MapgenFlat::MapgenFlat(MapgenFlatParams *params, EmergeParams *emerge)
 {
 	spflags            = params->spflags;
 	ground_level       = params->ground_level;
-	large_cave_depth   = params->large_cave_depth;
+	lake_threshold     = params->lake_threshold;
+	lake_steepness     = params->lake_steepness;
+	hill_threshold     = params->hill_threshold;
+	hill_steepness     = params->hill_steepness;
+
+	cave_width         = params->cave_width;
 	small_cave_num_min = params->small_cave_num_min;
 	small_cave_num_max = params->small_cave_num_max;
 	large_cave_num_min = params->large_cave_num_min;
 	large_cave_num_max = params->large_cave_num_max;
+	large_cave_depth   = params->large_cave_depth;
 	large_cave_flooded = params->large_cave_flooded;
-	cave_width         = params->cave_width;
-	lake_threshold     = params->lake_threshold;
-	lake_steepness     = params->lake_steepness;
-	hill_threshold     = params->hill_threshold;
-	hill_steepness     = params->hill_steepness;
+	cavern_limit       = params->cavern_limit;
+	cavern_taper       = params->cavern_taper;
+	cavern_threshold   = params->cavern_threshold;
 	dungeon_ymin       = params->dungeon_ymin;
 	dungeon_ymax       = params->dungeon_ymax;
```

Comment:
Most of this section of changes is moved lines, to separate this section of code into mapgen-specific parameters (ground level, lakes, hills) and shared code parameters (caves, caverns, dungeons).